### ISSUE TITLE
[BUG] It errors out on Xcode 16

### ISSUE THREAD

**Issue by marinofelipe** [ChargePoint/xcparse#87]
**Describe the bug**

The `xcresulttool` version that comes bundled with `Xcode 16` deprecates the commands currently used, such as `xcresulttool get object`, and introduces new ones.

For it to work normally, without changing to use the new commands, it  requires the `--legacy` argument to be passed when calling deprecated methods, otherwise it errors out, for example

```sh
mint run xcparse screenshots my_repo/someTarget.xcresult my_repo/output_dir
```
fails with
```
Error: Error: This command is deprecated and will be removed in a future release, --legacy flag is required to use it.
Usage: xcresulttool get object [--legacy] --path <path> [--id <id>] [--version <version>] [--format <format>]
  See 'xcresulttool get object --help' for more information.
``` 

**Desktop (please complete the following information):**
 - OS:  macOS 14.6.1 (23G93)
 - xcparse Version:  2.3.1 (latest)
 - Swift version: 6.0 (the default when using Xcode's 16(.1) toolchain)
 - XCResult version
 ```
 "version" : {
    "major" : 3,
    "minor" : 53
  }
 ```

**To Reproduce**
use `xcparse` with `Xcode 16` as the selected toolchain (i.e. via `xcode-select -s`)

**Expected behavior**
It should not error out

**Additional context**
I've a fork with a simple fix for it, let me know if I should move a PR with it, which I would have to ask a few follow up questions before opening it

Here's an example of a similar issue opened and addressed by the fastlane community: https://github.com/fastlane/fastlane/issues/22132

**Comment by rsukumar-cpi** [comment]
@marinofelipe Thanks for flagging the issue. You can go ahead and raise a PR. We will also look into updating the deprecated methods in the future.

CC: @abotkin-cpi

**Comment by megastep** [comment]
I'm running into this issue as well, and found a simple temporary workaround if you still have a copy of Xcode 15 installed on your system, to force the old version of `xcresulttool` being used.

Just set the `DEVELOPER_DIR` environment variable to the location of your Xcode 15 binary, i.e. `/Applications/Xcode15.app/Contents/Developer` before calling `xcparse`. This will only affect the version of Xcode being used while calling it when that environment variable is set, so you can still have your existing processes run with Xcode 16 except for extracting screenshots.

**Comment by abotkin-cpi** [comment]
We plan to release an updated xcparse next week that will include the legacy workaround once backwards compatibility with older Xcode toolchains is confirmed working. Thanks for bringing this up and working up a PR, @marinofelipe !

**Comment by marinofelipe** [comment]
This is now resolved in the 2.3.2 release, thank you all 🙇
